### ISSUE TITLE
Figure numbers in workbook pdf

### ISSUE THREAD

**Issue by mikeshulman** [active-calculus/active-calculus-single-mbx#285]
In the Workbook PDF (compiled locally from the master branch), many of the references to figures have the wrong number.

- Activity 1.2.4 refers to "Figure 1.2.11" but the figure is labeled 1.2.2.
- Preview Activity 1.3.1 refers to "Figure 1.3.2" but the figure is labeled 1.3.1.
- Activity 1.3.3 refers to "Figure 1.3.10" but the figure is labeled 1.3.2.
- and so on

**Comment by mikeshulman** [comment]
It looks like the generated LaTeX hardcodes the numbers in figure references, rather than sensibly using `\ref`.  Is this a general issue with pretext then?

**Comment by davidfarmer** [comment]
I am not familiar with this specific instance, but usually PreTeXt
has to insert all the reference numbers because otherwise there is no
way to have the same numbering in all the different output formats
(LaTeX/PDF, kindle, braille, HTML,...).

Also, \ref{} is not magic, because you also have to tell the LaTeX
the numbering scheme:  how many levels to number, for example.
And then you have to replicate that in other media.

Basically, don't look at the LaTeX source!  And if you were thinking
about editing the LaTeX, you have been warned.

**Comment by mikeshulman** [comment]
@davidfarmer Thanks for the explanation.  I have to look at the source to figure out what is causing this problem, though, and how it can be fixed.  I see why you want PreTeXt to insert all the reference numbers, but I think that in that case the issue is that PreTeXt is _not_ inserting the numbers on the figures themselves, but leaving that up to LaTeX.  For instance, in Activity 1.2.4 the reference text reads

```
...the graph in \hyperref{x:figure:F-1-2-act-pos]{Figure~1.2.11}
```

so PreTeXt has inserted the figure number there.  But the code inserting the figure itself reads

```
\begin{figureptx}{Plot of the position function \(y = s(t)\) in \hyperref[x:activity:act-1-2-3]{Activity~1.2.4}.}{x:figure:F-1-2-act-pos}{}
```

which doesn't put in the "1.2.11" anywhere.  So I think LaTeX is autogenerating a number for the figure from its internal counter, and in this case that number turns out to be different from what PreTeXt thought it should be because some intermediate figures have been omitted.

**Comment by davidfarmer** [comment]
It may be that there is a bug in PreTeXt, in which case it is a PreTeXt
issue, not an Active Calculus issue.

Even if it is a PTX bug, possibly a simple fix is to reference the
Activity and not the Figure in the Activity.

**Comment by mikeshulman** [comment]
Should I file an issue in PreTeXt, do you think?  I don't know enough about PreTeXt to do much more than point to this example.

In this case it is the text of the Activity itself which is referring to the Figure that it contains, so referring to the Activity wouldn't help.  (I suppose one might argue that it's not really necessary to refer to the figure by number at all in this case, that "the figure below" would suffice...)

**Comment by davidfarmer** [comment]
Maybe posting to the pretext-support group is a better option than
filling an issue.  https://groups.google.com/g/pretext-support

I personally like referring to "the accompanying figure" when that is 
unambiguous.

**Comment by rbeezer** [comment]
I can help with this, but we first need to hear from somebody on the Active Calculus side who can explain what stylesheets and what customizations are being used.   Subsetting is always tricky business, especially for LaTeX, and there are several ways to do it, including ad-hoc approaches.

Normally, studying the LaTeX is not advised.  Not much about `\ref{}` is sensible in LaTeX, so we have a variety of workarounds in place.  But we should begin with how **PreTeXt** is being employed, and that cannot be discerned from the LaTeX.

**Comment by mattboelkins** [comment]
Like Mitch, I don't have any bandwidth for this until early May.

Matt

**Comment by mikeshulman** [comment]
Makes sense.  In case anyone else is afflicted by this problem and would like a quick fix, after generating the workbook latex the following shell one-liner converts the hardcoded references to latex `\ref`s so that they generate the correct numbers when compiled:

```
sed -i -e 's/\\hyperref\[\(x:figure[^]]*\)\]{[^}]*}/Figure~\\ref{\1}/g' acs-activity-workbook-14.tex
```

**Comment by rbeezer** [comment]
Thanks for that incantation, @mikeshulman. We should get more serious about this once folks are done with classes.

**Comment by mattboelkins** [comment]
Returning to this thread, just to add one more:  Activity 1.1.4 refers to Activity 1.1.3 in HTML, but in the PDF workbook the reference is to Activity 1.1.2.  @mitchkeller let's talk about this in our meeting next week.

**Comment by mitchkeller** [comment]
> 

We apparently forgot to talk about this today. I think the main thing will be to check over the workbook references when we re-generate it.
